src/osd/PeeringState.cc

```diff
@@ -3086,6 +3086,13 @@ void PeeringState::merge_from(
 	       << sources.begin()->second->info.history
 	       << dendl;
 
+    if (info.history.epoch_created == 0) {
```

Comment:
We already check for this before getting here, see https://github.com/ceph/ceph/blob/master/src/osd/PeeringState.cc#L3071. Also, how about doing this adjustment for info.history.same_interval_since with https://github.com/ceph/ceph/blob/master/src/osd/PeeringState.cc#L3110-L3121, where the existing adjustment is done?

Reply:
> We already check for this before getting here, see https://github.com/ceph/ceph/blob/master/src/osd/PeeringState.cc#L3071.

That line is for merge target. If we hit that condition, merge target should be either a placeholder or a newly initiated peer that never got the chance to finish a complete peering cycle, then we'll try to steal info.history from merge source(we do that exact two lines below, see https://github.com/ceph/ceph/blob/master/src/osd/PeeringState.cc#L3074).
But the merge source could be a placeholder too, that's why we need to fix history.sis again.



> Also, how about doing this adjustment for info.history.same_interval_since with https://github.com/ceph/ceph/blob/master/src/osd/PeeringState.cc#L3110-L3121, where the existing adjustment is done?

That adjustment should happen only if we already have a meaningful history.same_interval_since in hand.

Reply:
makes sense, could you add this explanation as a comment for clarity